malfeasance2/handler.go

```diff
@@ -0,0 +1,140 @@
+package malfeasance2
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"go.uber.org/zap"
+
+	"github.com/spacemeshos/go-spacemesh/codec"
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/p2p"
+	"github.com/spacemeshos/go-spacemesh/p2p/pubsub"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+var (
+	ErrMalformedData  = fmt.Errorf("%w: malformed data", pubsub.ErrValidationReject)
+	ErrWrongHash      = fmt.Errorf("%w: incorrect hash", pubsub.ErrValidationReject)
+	ErrUnknownVersion = fmt.Errorf("%w: unknown version", pubsub.ErrValidationReject)
+	ErrUnknownDomain  = fmt.Errorf("%w: unknown domain", pubsub.ErrValidationReject)
+)
+
+type Handler struct {
+	logger   *zap.Logger
+	db       sql.Executor
+	self     p2p.Peer
+	tortoise tortoise
+
+	handlers map[ProofDomain]MalfeasanceHandler
+}
+
+func NewHandler(
+	db sql.Executor,
+	lg *zap.Logger,
+	self p2p.Peer,
+	tortoise tortoise,
+) *Handler {
+	return &Handler{
+		db:       db,
+		logger:   lg,
+		self:     self,
+		tortoise: tortoise,
```

Comment:
ideally we can avoid this dependency. why not have a pluggable function here that could be anything? and then you can just inject the tortoise hook that takes the reporting part. ideally we'd do the same in `malfeasance` package too